src/main/java/com/zendesk/maxwell/metrics/MaxwellHTTPServer.java

```diff
@@ -0,0 +1,64 @@
+package com.zendesk.maxwell.metrics;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.health.HealthCheckRegistry;
+import com.codahale.metrics.servlets.HealthCheckServlet;
+import com.codahale.metrics.servlets.MetricsServlet;
+import com.codahale.metrics.servlets.PingServlet;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.servlet.ServletContextHandler;
+import org.eclipse.jetty.servlet.ServletHolder;
+
+
+public class MaxwellHTTPServer {
+	public MaxwellHTTPServer(int port, MetricRegistry metricRegistry, HealthCheckRegistry healthCheckRegistry) {
+		MaxwellHTTPServerWorker maxwellHTTPServerWorker = new MaxwellHTTPServerWorker(port, metricRegistry, healthCheckRegistry);
+		Thread thread = new Thread(maxwellHTTPServerWorker);
+
+		thread.setUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
+			public void uncaughtException(Thread t, Throwable e) {
+				e.printStackTrace();
+				System.exit(1);
```

Comment:
i wondered the same. i can see both sides. i ended up here b/c if monitoring crashes you have significantly degraded visibility into what maxwell is doing.